### DISCUSSION TITLE
Cooktop Integration

### DISCUSSION THREAD

**Discussion by kapps7** [thoukydides/homebridge-homeconnect#321] [General]
Hi and thank you very much for the home bridge add on.

I have a cooktop BOSCH PIX631HC1E and it has been connected via the plugin into HomeKit.
I wasn't expecting to turn it on of course but I was hoping to turn it off. Is that an API limitation ?
If that's the case, what are people using the add on for. It seems that it's probably more useful for non cooktop appliances.

Thanks

**Comment by thoukydides** [comment]
The available functionality is listed in the project [README](https://github.com/thoukydides/homebridge-homeconnect?tab=readme-ov-file#appliance-support):

> * **Hob:**
>     * Control alarm clock and child lock.
>     * Monitor power, remote control, and general operation status.
>     * Automation triggers for finished, alarm clock finished, and preheat finished.

These are API limitations - supposedly for safety reasons. As you say, this plugin is more useful for other appliance types where the API does allow a useful level of control.

**Comment by thoukydides** [comment]
On the specific issue of controlling Cooktop/Hob power, this is what the API returns:
```
{
  "data": {
    "name": "Power status",
    "key": "BSH.Common.Setting.PowerState",
    "constraints": {
      "allowedvalues": [
        "BSH.Common.EnumType.PowerState.Off",
        "BSH.Common.EnumType.PowerState.On"
      ],
      "displayvalues": [
        "Off",
        "On"
      ],
      "access": "read"
    },
    "type": "BSH.Common.EnumType.PowerState",
    "displayvalue": "Off",
    "value": "BSH.Common.EnumType.PowerState.Off"
  }
}
```

Note the `"access": "read"` which indicates that the API only indicates the power status but doesn't allow it to be changed (it is `"readWrite"` for most other appliance types).

**Comment by kapps7** [comment]
Thanks and completely identify with your comments being API limitations. These ofcourse coincide with the Home Connect app.
No doubt they are in place for safety reasons but I find them counter intuitive.

eg. I would have thought that the ability to turn the power off remotely would enhance safety. Even their built in app notification is on when a timer schedule completes. If anything, the notification to alert that a hob is on would be more useful.

It would be interesting to know what automations people are triggering with the cooktop power. At most I can imagine a light perhaps in the kitchen where a home connect extractor is not integrated.
